src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -204,6 +217,8 @@ public void onSuccess(String accessToken) {
           if (!Objects.equals(currentUserAccessToken, accessToken)) {
             resetQueryClientCache();
             currentUserAccessToken = accessToken;
+          } else if (forceResetQueryClientCache) {
+            resetQueryClientCache();
           }
```

Comment:
Before you signed the ToU, we couldn't get your user profile. When you signed the ToU, your access token didn't change, so we didn't reset the QueryClient. So the useQuery instance to fetch your profile stayed in the `error` state.

Now, explicity force-reset the cache after signing the ToU.